### DISCUSSION TITLE
How to send client certificate from the IApiEndpointQueryContext / IServiceLocator ?

### DISCUSSION THREAD

**Discussion by ac2021-June** [unoplatform/uno#7966] [Q&A]
We want to send the client certificate from the application, to get the response from the webservice. Can you please help us with some sample code / guthub sample where we can get more information around it?

**Comment by jeromelaban** [comment]
`IApiEndpointQueryContext` is not part of Uno, so cannot help here. You may want to find this on the Xamarin-specific forums, like [this one](https://stackoverflow.com/questions/48951913/xamarin-android-adding-client-certificate), or with similar topic articles [like this one](https://nicksnettravels.builttoroam.com/ios-certificate/).

**Comment by limeniye** [comment]
I just know that you can use Xamarin.Essentials.WebAuthenticator.AuthenticateAsync(new Uri(url), new Uri(c.CallbackUrl))
I think its will not work in Uno Platfrom, but you can try.

**Comment by sasakrsmanovic** [comment]
One of our team members suggested to provide an IWebRequestCreate (https://docs.microsoft.com/en-us/dotnet/api/system.net.iwebrequestcreate?view=net-6.0) in the query : https://nventive.visualstudio.com/Umbrella/_git/Umbrella?path=/src/Umbrella.Query/Query/QueryExecution.cs&version=GBmaster&line=23&lineEnd=24&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

You should be able to configure certificate directly on the WebRequest using it cc @dr1rrb @carldebilly
